src/monster-floor/special-death-switcher.cpp

```diff
@@ -353,7 +354,7 @@ void switch_special_death(PlayerType *player_ptr, MonsterDeath *md_ptr, Attribut
         return;
     }
 
-    switch (md_ptr->m_ptr->r_idx) {
+    switch (md_ptr->ap_r_ptr->idx) {
```

Comment:
Since the summons on death is now unified and handled first, I think it is fine to keep the original here. Therefore, the decision to cancel the drop after this is also unnecessary.

Reply:
There are currently three monster types that have special at death behavior that isn't related to drops or summons:  Rolento, unmakers, and evil mani-mani.  The form of the switch statement in the pull request lets a chameleon acting as one of those three to mimic their at death behavior.